entity-server-api/src/main/java/org/terracotta/entity/CommonServerEntity.java

```diff
@@ -46,4 +46,12 @@
    *  respect to all other messages enqueued for the entity.</p>
    */
   void destroy();
+
+  /**
+   * Notify an entity that a particular client has a shared understanding of the state
+   * of the system, such as after the client disappears, or after reconnect has finished.
+   *
+   * @param client client
+   */
+  void notifyClientIsSynched(ClientDescriptor client);
```

Comment:
The action will be the same -- it will be called on disconnect (client isn't there anymore) and sometime after reconnect (when the client can be asserted to have caught up). In either case, you can throw away any tracking history, because we are asserting the client has the same picture as the server.